android/Samples/KMSample1/build.sh

```diff
@@ -1,61 +1,94 @@
 #!/usr/bin/env bash
-# Build KMSample1
+#
+# Samples: KMsample1
+
+set -eu
+# set -x: Debugging use, print each statement
+# set -x
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/../../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# This script runs from its own folder
+cd "$THIS_SCRIPT_PATH"
+
+################################ Main script ################################
+
+# Definition of global compile constants
+
+CONFIG="release"
+SAMPLE_FLAGS="build"
+
+builder_describe "Build KMSample1 app for Android." \
+  "@../../KMEA" \
+  "clean" \
+  "configure" \
+  "build" \
+  ":app                   KMSample1" \
+  "--ci                   Don't start the Gradle daemon. Use for CI" 
+
+# parse before describe_outputs to check debug flags
+builder_parse "$@"
+
+if builder_has_option --debug; then
+  builder_heading "### Debug config ####"
+  CONFIG="debug"
+  SAMPLE_FLAGS="assembleDebug"
+fi
 
-set -e
-set -u
+ARTIFACT="app-$CONFIG.apk"
 
-display_usage ( ) {
-    echo "build.sh [-no-daemon] [-debug]"
-    echo
-    echo "Build KM Sample 1"
-    echo "  -no-daemon              Don't start the Gradle daemon. Use for CI"
-    echo "  -debug                  Compile only Debug variant"
-    exit 1
-}
 
-echo Build KMSample1
+builder_describe_outputs \
+  configure             app/libs/keyman-engine.aar \
+  build:app             app/build/outputs/apk/$CONFIG/$ARTIFACT
 
 #
 # Prevents 'clear' on exit of mingw64 bash shell
 #
 SHLVL=0
 
-NO_DAEMON=false
-ONLY_DEBUG=false
 
 # Parse args
-while [[ $# -gt 0 ]] ; do
-    key="$1"
-    case $key in
-        -no-daemon)
-            NO_DAEMON=true
-            ;;
-        -debug)
-            ONLY_DEBUG=true
-            ;;
-        -h|-\?)
-            display_usage
-            ;;
-    esac
-    shift # past argument
-done
-
-echo
-echo "NO_DAEMON: $NO_DAEMON"
-echo "ONLY_DEBUG: $ONLY_DEBUG"
-echo
-
-if [ "$NO_DAEMON" = true ]; then
-  DAEMON_FLAG=--no-daemon
-else
-  DAEMON_FLAG=
+
+if builder_has_option --ci; then
+  SAMPLE_FLAGS="$SAMPLE_FLAGS -no-daemon"
 fi
 
-if [ "$ONLY_DEBUG" = true ]; then
-  BUILD_FLAG=assembleDebug
-else
-  BUILD_FLAG=build
+#### Build action definitions ####
+
+# Check about cleaning artifact paths and upload directories
+if builder_start_action clean; then
+  cd "$KEYMAN_ROOT/android/Samples/KMSample1/"
+
+  if [ -d "$KEYMAN_ROOT/android/Samples/KMSample1/app/build/outputs" ]; then
+    rm -rf "$KEYMAN_ROOT/android/Samples/KMSample1/app/build/outputs"
+  fi
+
+  if [ -d "$KEYMAN_ROOT/android/upload" ]; then
+    rm -rf "$KEYMAN_ROOT/android/upload"
+  fi
```

Comment:
```suggestion
  rm -rf "$KEYMAN_ROOT/android/Samples/KMSample1/app/build/outputs"
```

The top-level upload folder should not be deleted by any of the child scripts. See also all the other builder scripts